### PR TITLE
Clean up nulls in unit tests. Reduce the 300 "Nullable value type may be null" warnings

### PR DESCRIPTION
**Pull request by jsimonsmartsheet**
Previously on mainline 
```
git clean -xfd && dotnet build
...
   301 Warning(s)
    0 Error(s)
```

now 
```
git clean -xfd && dotnet build
...
    59 Warning(s)
    0 Error(s)
 ```
 
 this MR also bumps the vulnerable version of the restsharp sdk

### CODE REVIEW

File: smartsheet-csharp-sdk/main/Smartsheet/Api/SheetResources.cs [hunk 1]
```diff
@@ -91,6 +91,9 @@ public interface SheetResources
         /// <param name="columnIds"> used to specify the optional objects to include. </param>
         /// <param name="pageSize"> used to specify the optional objects to include. </param>
         /// <param name="page"> used to specify the optional objects to include. </param>
+        /// <param name="rowsModifiedSince">FIXME: WHAT IS THIS PARAM</param>
```
Comment (jsimonsmartsheet) [inline smartsheet-csharp-sdk/main/Smartsheet/Api/SheetResources.cs]: what should I include here?

File: sdk-csharp-sample-net80/sdk-csharp-sample-net80.csproj [hunk 2]
```diff
@@ -22,7 +22,7 @@
   <ItemGroup>
     <PackageReference Include="System.IO" Version="4.3.0" />
     <PackageReference Include="System.Threading" Version="4.3.0" />
-    <PackageReference Include="RestSharp" Version="109.0.0" />
+    <PackageReference Include="RestSharp" Version="112.0.0" />
```
Comment (timothodge) [inline sdk-csharp-sample-net80/sdk-csharp-sample-net80.csproj]: I think we may want to put this in it's own PR. The jump to 109 was chaotic and I think we may want to give customers the heads up about that and put it in its own release.

File: smartsheet-csharp-sdk/main/Smartsheet/Api/SheetResources.cs (hunk 1 above)
Comment (timothodge) [reply smartsheet-csharp-sdk/main/Smartsheet/Api/SheetResources.cs]: I might consider pulling in mainline. I don't see these on mainline.... ?

File: sdk-csharp-sample-net80/sdk-csharp-sample-net80.csproj (hunk 2 above)
Comment (jsimonsmartsheet) [reply sdk-csharp-sample-net80/sdk-csharp-sample-net80.csproj]: ok, reverted

File: smartsheet-csharp-sdk/main/Smartsheet/Api/SheetResources.cs (hunk 1 above)
Comment (jsimonsmartsheet) [reply smartsheet-csharp-sdk/main/Smartsheet/Api/SheetResources.cs]: I see the yellow lines under the parameters to GetSheet. But I can pull them for now.
image

**Review by timothodge — APPROVED** [review]
Looks good thanks for knocking down the warnings!
